server/aesdsocket.c

```diff
@@ -71,47 +164,262 @@ static void sig_handler(int signo){
         syslog(LOG_ERR,"\nERROR in deleting file");
     }
     
-    // close log
-    closelog();
+
     
-    // free pointers
-    free(read_buf);
-    free(write_buf);
+    // Cancel threads &free pointers
+    SLIST_FOREACH(datap,&head,entries){
+
+        if (datap->threadParams.completion_status != true){
+
+            pthread_cancel(datap->threadParams.threaddec);
+            free(datap->threadParams.read_buf);
+            free(datap->threadParams.write_buf);
+            
+        }
+
+
+    }
     
-    exit(0);
+    // free Linked list
+    while(!SLIST_EMPTY(&head)){
+        datap = SLIST_FIRST(&head);
+        SLIST_REMOVE_HEAD(&head,entries);
+        free(datap);
+    }
 
+    pthread_mutex_destroy(&file_mutex);                       // destroy mutex
 
+    timer_delete(timerid);                                    // delete timer
```

Comment:
@shni9045 what would happen if a timer event occurred between `pthread_mutex_destroy` and `timer_delete`?  What would happen when your timer attempted to access `file_mutex`?